druzhba/table.py

```diff
@@ -637,12 +637,17 @@ def where_clause(self):
             # Either the table is empty or the index_column is all NULL
             return ""
 
-        if self.old_index_value:
-            # This should always happen except on the initial load
+        if self.old_index_value and not self.full_refresh:
+            # This should always happen except on the initial load or if a
+            # full_refresh is explicitly requested. The old_index_value should
+            # be None in the full_refresh case but this adds an extra guard.
             where_clause += "{} > '{}' AND ".format(
                 self.index_column, self.old_index_value
             )
 
+        # Note that we include the new_index_value as an upper bound even in
```

Comment:
There are two ways `full_refresh` can be set: 
- as a CLI parameter
- or in the configuration for a table

In the latter case, there cannot be an index column at all (configuration check says:
```
        elif index_column and full_refresh:
            raise ConfigurationError("Full refresh with index_column", table)
```
)
which means the impl of `new_index_value` will give `None` and this `where` will be ill-formed. 

So line 651 here needs to check that `index_column` exists, or something else to handle this case

Reply:
Ahh, thanks for calling this out. I believe the `if self.new_index_value is None: return ''` check on 636 should handle that case but I will make sure to test with a `full_refresh` table.

Reply:
you're right it should - I didn't see that line.